robot/web/web.go

```diff
@@ -564,6 +566,55 @@ func (rc *RequestCounter) UnaryInterceptor(
 	return handler(ctx, req)
 }
 
+type wrappedStreamWithRC struct {
+	googlegrpc.ServerStream
+	apiMethod string
+	rc        *RequestCounter
+}
+
+func (w *wrappedStreamWithRC) RecvMsg(m any) error {
+	// Unmarshalls into m (to populate fields).
+	err := w.ServerStream.RecvMsg(m)
+
+	if err == nil && w.apiMethod != "" {
+		var key string
+		if namer, ok := m.(Namer); ok {
+			key = fmt.Sprintf("%v.%v", namer.GetName(), w.apiMethod)
+		} else {
+			key = w.apiMethod
+		}
+		if apiCounts, ok := w.rc.counts.Load(key); ok {
+			apiCounts.(*atomic.Int64).Add(1)
+		} else {
+			newCounter := new(atomic.Int64)
+			newCounter.Add(1)
+			w.rc.counts.Store(key, newCounter)
+		}
+	}
+
+	return err
+}
+
+func (rc *RequestCounter) StreamInterceptor(
+	srv any,
+	ss googlegrpc.ServerStream,
+	info *googlegrpc.StreamServerInfo,
+	handler googlegrpc.StreamHandler,
+) error {
+	var apiMethod string
+	switch {
+	case strings.HasPrefix(info.FullMethod, "/viam.component."):
+		fallthrough
+	case strings.HasPrefix(info.FullMethod, "/viam.service."):
+		fallthrough
+	case strings.HasPrefix(info.FullMethod, "/viam.robot."):
+		apiMethod = info.FullMethod[strings.LastIndexByte(info.FullMethod, byte('.'))+1:]
+	default:
+	}
+
+	return handler(srv, &wrappedStreamWithRC{ss, apiMethod, rc})
```

Comment:
is wrappedStreamWithRC necessary? can we put the logic in `RecvMsg` here and call the handler without wrapping the stream in a new type?